### PR TITLE
[service.onidle] 0.0.4

### PR DESCRIPTION
**Pull request by Lunatixz**
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by Lunatixz** [comment]
Travis False Positive.... The entry point limit warning is for a service.

File: service.onidle/addon.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="service.onidle" version="0.0.4" name="OnIdle" provider-name="Lunatixz">
+    <requires>
+        <import addon="xbmc.python" version="2.26.0"/>
+    </requires>
+    <extension point="xbmc.python.script" library="settings.py"/>
+    <extension point="xbmc.service" library="service.py" start="startup"/>
```
Comment (enen92) [inline service.onidle/addon.xml]: @Lunatixz this is not a false positive: https://github.com/xbmc/addon-check/commit/33c3fa1618d3c42088212b4bd202115b4022fafd

File: service.onidle/addon.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    <extension point="xbmc.addon.metadata">
+        <summary lang="en_GB">Idle Sleep Timer</summary>
+        <description lang="en_GB">Trigger Kodi actions when idle while playing.</description>
+        <license>GNU GENERAL PUBLIC LICENSE. Version 3, June 2007</license>
```
Comment (enen92) [inline service.onidle/addon.xml]: Use a SPDX identifier for the license please: https://spdx.org/licenses/
I'll had this to the checker at some point

File: service.onidle/resources/settings.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes"?>
```
Comment (enen92) [inline service.onidle/resources/settings.xml]: Would be nice to migrate this to the new settings format at some point in the future

File: service.onidle/resources/settings.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    <setting id="Enable_Debugging"   type="bool"   label="32001"  default="false" />
+    <setting id="Manual_Run"         type="bool"   label="32020"  default="false" enable="false" visible="false"/>
+    <setting id="Ignore_Music"       type="bool"   label="32005"  default="true" />
+    <setting id="Soft_Mute"          type="bool"   label="32016"  default="true" />
+    <setting id="User_Idle_Min"      type="slider" label="32003"  default="30" range="30,15,900" option="int" />
+    <setting id="User_Countdown_Sec" type="slider" label="32004"  default="30" range="30,15,300" option="int" />
+    <setting id="User_Exit_Action"   type="enum"   label="32002"  lvalues="32013|32012|32011|32010|32009|32008|32007|32006"/>
+    <setting id="User_Action"        type="text"   label="32015"  default="" subsetting="true" visible="eq(-1,7)" />
+    <setting id="PROMPT"             type="text"   label="32014"  enable="false" visible="eq(-2,7)" />
+    <setting id="EXE_User_Action"    type="action" label="32021"  action="RunScript($CWD/test.py)" option="close" />
```
Comment (enen92) [inline service.onidle/resources/settings.xml]: CWD is deprecated I guess, please invoke your script path. E.g: `RunScript(script.id, args)`

File: service.onidle/service.py [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+    def checkShutdown(self, idleTime):
+        log('checkShutdown')
+        if USER_IGNORE_MUSIC and self.myPlayer.isPlayingAudio(): return False
+        diaProgress = xbmcgui.DialogProgress()
+        ret         = diaProgress.create(ADDON_NAME,LANGUAGE(32017))
+        secs        = 0
+        percent     = 0
+        increment   = 100*100 / USER_COUNTDOWN
+        cancelled   = False
+        while secs < USER_COUNTDOWN:
```
Comment (enen92) [inline service.onidle/service.py]: abort requested must be added

**Pull request merged**